add-ons/istio/README.adoc

```diff
@@ -0,0 +1,52 @@
+:linkattrs:
+
+= Istio Add-on
+
+An addon that will deploy https://Istio.io/[istio] service mesh on to Openshift.
+
+Verify you have installed these addons, by following the link:../../README.adoc#download-and-use-community-add-ons[general readme]
+
+== minishift profile setup
+
+As https://Istio.io/[istio] setup spins up multiple pods its ideal to have a setup with the following configuration
+
+[code,sh]
+----
+minishift profile set servicemesh <1>
+minishift config set memory 8192 <2>
+minishift config set cpus 3 <3>
+minishift addon enable admin-user <4>
+minishift addon enable anyuid <5>
+minishift config set openshift-version v3.7.0 <6>
+----
+<1> create a new profile say servicemesh
+<2> set a memory of 8192, this is good amount of memory for istio + apps that you might deploy later
+<3> since Istio minimum deploys approximately __9__ pods, 3 cpus will give a max of 30 pods
+<4> Admin user addon allows you to login with user admin/admin, who has admin privileges on all openshift projects
+<5> since some of the Istio pods requires to be run as ANYUID, this sets the openshift security context to allow the pod to run with anyuid
+<6> since Istio 0.2.x and above can be deployed with Kubernetes 1.7.x, its requried that OpenShift version 3.7.x or above
+
+== Download 
+Download Istio corresponding to your platform from https://github.com/istio/istio/releases
+extract it to a folder of your choice and export that folder as `ISTIO_HOME`
+
+== Deploy Istio
+
+[code,sh]
+----
+minishift addon apply istio --addon-env ISTIO_HOME=$ISTIO_HOME
+----
+
+== Undeploy Istio
+
+[code,sh]
+----
+minishift remove istio
+----
+
+== Undeploy Istio addon
```

Comment:
This step is not necessary. It will remove the add-on artifacts from the system.

Reply:
yes i want to show the user about it as well hence purposefully did it